src/main.rs

```diff
@@ -73,6 +77,42 @@ macro_rules! unwrap_or_show_error {
     }};
 }
 
+fn get_mirror_url() -> Result<String> {
+    let mut status_file = File::open("/var/lib/apt/gen/status.json")?;
+    let mut status_data = String::new();
+    status_file.read_to_string(&mut status_data)?;
+
+    let status_v: jsonValue = serde_json::from_str(&status_data)?;
+    let mirror: String = match &status_v["mirror"][0] {
+        jsonValue::String(m) => String::from(m),
+        _ => {
+            return Err(format_err!("Malformed apt-gen-list data."));
+        }
+    };
```

Comment:
There is a better way to do this deserialize song and dance, take a look in https://github.com/AOSC-Dev/atm/blob/master/src/network.rs?